pkg/graph/constructor.go

```diff
@@ -17,14 +17,155 @@ limitations under the License.
 package graph
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 
+	"go.uber.org/zap"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	rest "k8s.io/client-go/rest"
+
 	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
 	eventingv1beta3 "knative.dev/eventing/pkg/apis/eventing/v1beta3"
 	messagingv1 "knative.dev/eventing/pkg/apis/messaging/v1"
+	eventingclient "knative.dev/eventing/pkg/client/clientset/versioned"
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 )
 
+type ConstructorConfig struct {
+	RestConfig            rest.Config
+	Namespaces            []string
+	ShouldAddBroker       func(b eventingv1.Broker) bool
+	ShouldAddChannel      func(c messagingv1.Channel) bool
+	ShouldAddSource       func(s duckv1.Source) bool
+	ShouldAddTrigger      func(t eventingv1.Trigger) bool
+	ShouldAddSubscription func(s messagingv1.Subscription) bool
+	ShouldAddEventType    func(et eventingv1beta3.EventType) bool
```

Comment:
These are nice, but the resources are fetched anyway. If I am not interested in channels, that shouldn't be fetched at all.

Can you create separate entries like `fetchBrokers`, etc defaulting to true?